client/src/components/dashboard.js

```diff
@@ -75,6 +78,9 @@ function Dashboard(props) {
             </tbody>
           </table>
         ))}
+      {joinedChannel ? (
+        <Redirect to={`/channels/${joinedChannel.name}`} />
```

Comment:
A best practice is to use redirect only on Route `render` methods instead of inside the component. We can do this by using `props.history.push`, passing history as a prop from `<Route render={(props) => ...}/>`